crates/matrix-sdk/src/media.rs

```diff
@@ -96,6 +113,43 @@ impl Media {
         Ok(self.client.send(request, Some(request_config)).await?)
     }
 
+    /// Gets a media file by copying it to a temporary location on disk.
+    ///
+    /// If the content is encrypted and encryption is enabled, the content will
+    /// be decrypted.
+    ///
+    /// Returns a `MediaFileHandle` which takes ownership of the file. When the
+    /// handle is dropped, the file will be deleted from the temporary location.
+    ///
+    /// # Arguments
+    ///
+    /// * `request` - The `MediaRequest` of the content.
+    ///
+    /// * `content_type` - The type of the media, this will be used to set the
+    ///   temporary file's extension.
+    ///
+    /// * `use_cache` - If we should use the media cache for this request.
+    pub async fn get_media_file(
+        &self,
+        request: &MediaRequest,
+        content_type: &Mime,
+        use_cache: bool,
+    ) -> Result<MediaFileHandle> {
+        let data = self.get_media_content(request, use_cache).await?;
+
+        let mut suffix = String::from("");
+        if let Some(extension) =
+            mime_guess::get_mime_extensions(content_type).and_then(|a| a.first())
+        {
+            suffix = String::from(".") + extension;
+        }
+
+        let mut file = TempFileBuilder::new().suffix(&suffix).tempfile()?;
+        file.write_all(&data)?;
```

Comment:
We shouldn't use blocking I/O in an `async` function. I think you can use `NamedTempFile::reopen` + `tokio::fs::File::from_std` to write to it in an async way. Or you can use `tokio::task::spawn_blocking(/* ... */).await`, but I think that will require `reopen` as well to satisfy lifetime constraints.